test/suite/test_timestamp04.py

```diff
@@ -44,103 +44,244 @@ def timestamp_ret_str(t):
+        latest_ts = timestamp_str(2 * key_range)
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_nots_log, dict((k, 2) for k in keys[:]))
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_nots_nolog, dict((k, 2) for k in keys[:]))
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_ts_log, dict((k, 2) for k in keys[:]))
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_ts_nolog, dict((k, 2) for k in keys[:]))
 
-        # Rollback a quater of the new commits
-        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(1 + key_range + key_range / 4))
+        #Scenario: 4
+        # Advance the stable_timestamp by a quater range and rollback.
+        # three-quarter timestamps will be rolledback.
+        stable_ts = timestamp_str(1 + key_range + key_range / 4)
+        self.conn.set_timestamp('stable_timestamp=' + stable_ts)
         self.conn.rollback_to_stable()
 
-        # There should be 50 keys, the first half of which have a value of 2, the
-        # second half have a value of 1
-        self.check(self.session, 'read_timestamp=' + timestamp_str(2 * key_range),
-            dict((k, (2 if j <= (key_range / 4) else 1))
+        # Check that we see the updated value (i.e. 2) for all the keys in
+        # non-timestamp tables
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_nots_log, dict((k, 2) for k in keys[:]))
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_nots_nolog, dict((k, 2) for k in keys[:]))
+        # if nologging tables the behavior is consistent across connections
+        # with or without log enabled
+        # Check that we see only half key ranges in timestamp tables.
+        # Check that we see the updated value (i.e. 2) for the first quarter
+        # keys in a timestamp table and old values (i.e. 1) for the second
+        # quarter keys.
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_ts_nolog, dict((k, (2 if j <= (key_range / 4) else 1))
             for j, k in enumerate(keys[:(key_range / 2)])))
-        self.check(self.session, 'read_timestamp=' + timestamp_str(key_range / 2),
-            dict((k, 1) for k in keys[(1 + key_range / 2):]), missing=True)
+        self.check(self.session, 'read_timestamp=' + latest_ts,
+            self.table_ts_nolog, dict((k, 1) for k in keys[(1 + key_range / 2):]), missing=True)
+
+        # behavior of logging tables changes for rollback
+        if self.using_log == True:
+            # if log is enabled, none of the keys will be rolledback.
+            # Check that we see all the keys
+            self.check(self.session, 'read_timestamp=' + latest_ts,
+                self.table_ts_log, dict((k, 2) for k in keys[:]))
+        else:
```

Comment:
Aren't these tests same as the tests starting at line # 262

Reply:
initial tests are for inserts, the second round covers updates aswell.